azurerm/resource_arm_app_service_slot.go

```diff
@@ -186,46 +188,107 @@ func resourceArmAppServiceSlotCreateUpdate(d *schema.ResourceData, meta interfac
 		siteEnvelope.Identity = appServiceIdentity
 	}
 
-	createFuture, err := client.CreateOrUpdateSlot(ctx, resGroup, appServiceName, siteEnvelope, slot)
+	createFuture, err := client.CreateOrUpdateSlot(ctx, resourceGroup, appServiceName, siteEnvelope, slot)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error creating Slot %q (App Service %q / Resource Group %q): %s", slot, appServiceName, resourceGroup, err)
 	}
 
 	err = createFuture.WaitForCompletionRef(ctx, client.Client)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error waiting for creation of Slot %q (App Service %q / Resource Group %q): %s", slot, appServiceName, resourceGroup, err)
 	}
 
-	read, err := client.GetSlot(ctx, resGroup, appServiceName, slot)
+	read, err := client.GetSlot(ctx, resourceGroup, appServiceName, slot)
 	if err != nil {
-		return err
+		return fmt.Errorf("Error retrieving Slot %q (App Service %q / Resource Group %q): %s", slot, appServiceName, resourceGroup, err)
 	}
 
 	if read.ID == nil {
-		return fmt.Errorf("Cannot read App Service Slot %q/%q (resource group %q) ID", appServiceName, slot, resGroup)
+		return fmt.Errorf("Cannot read ID for Slot %q (App Service %q / Resource Group %q) ID", slot, appServiceName, resourceGroup)
 	}
 
 	d.SetId(*read.ID)
 
+	return resourceArmAppServiceSlotUpdate(d, meta)
+}
+
+func resourceArmAppServiceSlotUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	appServiceName := id.Path["sites"]
+	slot := id.Path["slots"]
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	appServicePlanId := d.Get("app_service_plan_id").(string)
+	siteConfig := azure.ExpandAppServiceSiteConfig(d.Get("site_config"))
+	enabled := d.Get("enabled").(bool)
+	httpsOnly := d.Get("https_only").(bool)
+	tags := d.Get("tags").(map[string]interface{})
```

Comment:
Should the outcome of this assertion be tested for potential errors?

Reply:
nah - we do this throughout the provider since the schema's shared - such that whilst this /could/ be an issue, the entire provider would fail if something was wrong; such that this is fine/would be caught by the tests 👍